app/components/input/input-form.tsx

```diff
@@ -7,20 +8,24 @@ import {
   forwardRef,
   useColorModeValue
 } from '@chakra-ui/react'
-
+import { FiEye, FiEyeOff } from 'react-icons/fi'
 type InputFormProps = {
   rightIcon?: IconType
+  hasEyeIcon?: boolean
 }
 export const InputForm = forwardRef<ChakraInputProps & InputFormProps, 'input'>(
-  ({ rightIcon: RightIcon, ...props }, ref) => {
+  ({ rightIcon: RightIcon, hasEyeIcon, ...props }, ref) => {
     const bgColor = useColorModeValue('white', 'gray.700')
     const borderColor = useColorModeValue('gray.300', 'gray.500')
     const placeholdercolor = useColorModeValue('gray.400', 'gray.600')
+    const [show, setShow] = useState(false)
+    const handleClick = () => setShow(!show)
 
     return (
       <InputGroup minW={'100%'}>
         <ChakraFormInput
           {...props}
+          type={show ? 'text' : props.type}
```

Comment:
There was a mistake in my code. It should pass a defined input type if 'show' is not true. Previously, it was only passing the type 'password,' causing all input fields to become of type password.